DOC Improve docs of permutation importance on the user guide (#27154)
scikit-learn/scikit-learn: doc/modules/permutation_importance.rst

.. currentmodule:: sklearn.inspection

Permutation feature importance is a model inspection technique that measures the

> contribution of each feature to a :term:`fitted` model's performance on a given

Reviewer: when using performance I think that we should make it explicit that we mean statictical performance.

tabular dataset. This technique is particularly useful for non-linear or opaque
:term:`estimators`, and involves randomly shuffling the values of a single
feature and observing the resulting degradation of the model's score [1]_. By

> breaking the relationship between the feature and the target, we can determine

Reviewer: ```suggestion
breaking the relationship between the feature and the target, we determine
```

> how much the model relies on such feature. One key advantage of permutation

Reviewer: ```suggestion
how much the model relies on this particular feature. One key advantage of permutation
```

feature importance is that it is model-agnostic, i.e. it can be applied to any
fitted estimator. Moreover, it can be calculated multiple times with different

> permutations of the feature, providing a robust measure of feature importance

Reviewer: I would not use "robust" but instead phrase it that we have an estimate of the results variance.

for the specific trained model.

The figure below shows the permutation feature importance of a
:class:`~sklearn.ensemble.RandomForestClassifier` trained on an augmented

> version of the titanic dataset that contains a `random_cat` and a `random_num`

Reviewer: At this stage, the reader does not know what `random_cat` and `random_num` are. I think that we can mention that we included two random features in the dataset and refer to the feature names then.

.. currentmodule:: sklearn.inspection

Permutation feature importance is a model inspection technique that measures the
contribution of each feature to a :term:`fitted` model's statistical performance
on a given tabular dataset. This technique is particularly useful for non-linear
or opaque :term:`estimators`, and involves randomly shuffling the values of a
single feature and observing the resulting degradation of the model's score
[1]_. By breaking the relationship between the feature and the target, we
determine how much the model relies on such particular feature.

> In the following figures we show how permutation of a predictive feature breaks
> the correlation, whereas permutation of a non-predictive feature does not
> considerably degrade the learned model.
> 
> .. image:: ../images/permuted_predictive_feature.png
>    :align: center
> 
> .. image:: ../images/permuted_non_predictive_feature.png
>    :align: center

Reviewer: I would provide the explanation after the figures.

```suggestion
In the following figures, we should the effect of permuting features on the correlation
between the feature and the target and consequently on the model statistical
performance.

.. image:: ../images/permuted_predictive_feature.png
   :align: center

.. image:: ../images/permuted_non_predictive_feature.png
   :align: center

On the top figure, we observe that permuting a predictive feature breaks the
correlation between the feature and the target, and consequently the model
statistical performance decreases. On the bottom figure, we observe that permuting
a non-predictive feature does not degrade the model statistical performance.
```

inspected model. Features that are important on the training set but not on the
held-out set might cause the model to overfit.

> The permutation feature importance depends on the score function that can be

Reviewer: ```suggestion
The permutation feature importance depends on the score function that is
```

> specified with the `scoring` argument. Said argument accepts multiple scorers,

Reviewer: ```suggestion
specified with the `scoring` argument. This argument accepts multiple scorers,
```

The ranking of the features is approximately the same for different metrics even
if the scales of the importance values are very different. However, this is not

> guaranteed and different metrics might lead to significantl*y different feature

Author: ```suggestion
guaranteed and different metrics might lead to significantly different feature
```

-------------------------------------------------

When two features are correlated and one of the features is permuted, the model
still has access to the latter through its correlated feature. This results in a

> lower importance value for both features, though they might *actually* be

Reviewer: ```suggestion
lower reported importance value for both features, though they might *actually* be
```
